### PR TITLE
Image uploader mime filter uses urls but doesn't check them

### LINKED ISSUE #1688: Pasting text from JetBrains IDEs tries to upload the text to i.nuuls.com instead of pasting text
**Describe the bug**

**To reproduce**
1. copy text from a JetBrains IDE (IntelliJ IDEA, Android Studio, PyCharm, etc...) - text can be from the terminal, the editor area, or the execution console output, and probably other places too
2. Paste the text into the chatterino input box
3. Popup for nuuls image uploader opens and no text is pasted into the input box

**Screenshots**
https://streamable.com/44fcqk
image

**Chatterino version**
Chatterino Nightly 2.1.7 (commit d9aab1f6e, ) 

**Operating system**
Windows 10. Seems to only affect Windows

**Additional information**

Was confirmed to affect another user, using Android Studio.

### PR DESCRIPTION
**Pull request by Mm2PL**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Attempt to fix #1688 @randers00 please check if it works BabyRage

### CODE REVIEW

File: src/widgets/helper/ResizingTextEdit.cpp [hunk 1]
```diff
@@ -264,14 +265,32 @@ bool ResizingTextEdit::canInsertFromMimeData(const QMimeData *source) const
...
     {
-        insertPlainText(source->text());
+        bool hasUploadable = false;
+        auto mimeDb = QMimeDatabase();
+        for (const QUrl url : source->urls())
+        {
+            QMimeType mime = mimeDb.mimeTypeForUrl(url);
+            if (mime.name().startsWith("image"))
+            {
+                hasUploadable = true;
```
Comment (pajlada) [inline src/widgets/helper/ResizingTextEdit.cpp:282]: Is the logic "If the list of urls contains at minimum one image url, upload it?" - if so, we could early out here.
```suggestion
                hasUploadable = true;
                break;
```

**Comment by RAnders00** [comment]
I will test this out tomorrow

**Review by RAnders00 — APPROVED** [review]
Works for me!

**Review by pajlada — APPROVED** [review]
With @RAnders00 's approval I approve this! Thanks for the fix!

**Pull request merged**